geomstats/geometry/pre_shape.py

```diff
@@ -378,6 +379,339 @@ def integrability_tensor(self, tangent_vec_a, tangent_vec_b, base_point):
+        (A_Y A_X Y) - A_X A_Y A_X Y` (where :math: `X`and :math: `Y` are
+        horizontal vector fields) is a key ingredient in the computation of
+        the covariant derivative of the directional curvature in a submersion.
+        The components :math: `\nabla_X^S (A_Y A_X Y)`, :math: `A_X A_Y A_X Y`,
+        :math: `\nabla_X^S (A_X Y)`,  and intermediate computations
+        :math: `A_Y A_X Y` and :math: `A_X Y` are computed here for the
+        Kendall shape space in the special case of quotient-parallel vector
+        fields :math: `X, Y` extending the values tg_vec_x and tg_vec_y by
+        parallel transport in a neighborhood. Such vector fields verify
+        :math: `\nabla^S_X^X = A_X X` and :math: `\nabla^S_X^Y = A_X Y`.
+
+        Parameters
+        ----------
+        tg_vec_x : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`.
+        tg_vec_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`.
+        base_point : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Point of the total space.
+
+        Returns
+        -------
+        nabla_x_a_y_a_x_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`, result of :math:
+            `\nabla_X^S (A_Y A_X Y)` with `X = tg_vec_x` and `Y = tg_vec_y`.
+        a_x_a_y_a_x_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`, result of :math:
+            `A_X A_Y A_X Y` with `X = tg_vec_x` and `Y = tg_vec_y`.
+        nabla_x_a_x_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`, result of :math:
+            `\nabla_X^S (A_X Y)` with `X = tg_vec_x` and `Y = tg_vec_y`.
+        a_y_a_x_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`, result of :math:
+            `A_Y A_X Y` with `X = tg_vec_x` and `Y = tg_vec_y`.
+        a_x_y : array-like, shape=[..., {ambient_dim, [n, n]}]
+            Tangent vector at `base_point`, result of :math:
+            `A_X Y` with `X = tg_vec_x` and `Y = tg_vec_y`.
+
+        References
+        ----------
+        X Pennec. To be published.
+        """
+        # Vectors X and Y have to be horizontal.
+        if not gs.all(self.is_centered(base_point)):
+            raise ValueError('The base_point does not belong to the pre-shape'
+                             ' space')
+        if not gs.all(self.is_horizontal(tg_vec_x, base_point)):
+            raise ValueError('Tangent vector x is not horizontal')
+        if not gs.all(self.is_horizontal(tg_vec_y, base_point)):
+            raise ValueError('Tangent vector y is not horizontal')
+        # hor_x = self.horizontal_projection(tangent_vec_x, base_point)
+        # hor_y = self.horizontal_projection(tangent_vec_y, base_point)
+
+        p_top = Matrices.transpose(base_point)
```

Comment:
Or not... base_point_top_base_point might be a bit long indeed...

Reply:
Well pushing the logic, we should actually use transposed instead of to, so that `base_point_transposed_base_point` would already takes half a line if not all with the indentation. We could also use `horizontal_vec_x_vertical_vec_v_transposed` instead of `x_v_top`, but just writing `horizontal_vec_x_vertical_vec_v_transposed = gs.matmul(vertical_v_transposed, horizontal_vec_x)` would take two lines (if not more), and I do not think that it would help anyone to better understand what is done here. 

I had to fight heavily with the 79 character limit on each line to keep something readable. I should also stress that it took me 2 week to spot a bug in the computations due to the use of `tangent_vec_x` in place of `tangent_vec_y` (one letter changing among 13) while it would have been obvious if the variable x and y were used.  

There is a point were adding very long variable names like that is just hiding the mathematical operation done without giving a hint about what it to the other users/maintainers. I don't think there is a way to make sense of this internal code without the latex file describing the computations. Getting away from the latex notations for such complex calculus is according to me the best way to obtain a cryptic code which cannot be debugged any more.